docs/getting-started-guides/locally.md

```diff
@@ -129,6 +132,19 @@ cluster/kubectl.sh create -f docs/user-guide/pod.yaml
 
 Congratulations!
 
+### Hacking hack-local-up cluster to test more sophisticated deployments, such as HA.
+
+To test highly available apiservers, you can modify local-up-cluster.sh as follows.
+
+1) Write a kubernetes token file which has a kubelet user with a token defined.  Lets call it /tmp/mytokens for concreteness.
+
+2) Launch the apiserver with the `--token-auth-file=/tmp/mytokens` and `--apiserver-node-port=30279` options.  Of course 30279 is just an example port.
```

Comment:
You additionally need a load balancer for that, and using that address from
within the cluster would be an extra hop, even assuming you're OK with the
egress/ingress.
On Sep 23, 2015 6:57 PM, "jay vyas" <notifications@github.com> wrote:

> In docs/getting-started-guides/locally.md
> <https://github.com/kubernetes/kubernetes/pull/13978#discussion_r40279262>
> :
>
> > @@ -129,6 +132,19 @@ cluster/kubectl.sh create -f docs/user-guide/pod.yaml
> >
> >  Congratulations!
> >
> > +### Hacking hack-local-up cluster to test more sophisticated deployments, such as HA.
> > +
> > +To test highly available apiservers, you can modify local-up-cluster.sh as follows.
> > +
> > +1) Write a kubernetes token file which has a kubelet user with a token defined.  Lets call it /tmp/mytokens for concreteness.
> > +
> > +2) Launch the apiserver with the `--token-auth-file=/tmp/mytokens` and `--apiserver-node-port=30279` options.  Of course 30279 is just an example port.
>
> AFAIK the original thought was that this would be a nice clean universal
> url that could be used by anybody to access apiservers, but I agree there
> isn't a functional improvement when using the url if you're already in the
> ViP.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13978/files#r40279262>.
>